src/components/CheckoutButton.tsx

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { loadStripe } from "@stripe/stripe-js";
+import { Button } from "@chakra-ui/react";
+
+// stripePromise object which is resolved in the component
+// for redirecting to checkout
+const stripePromise = loadStripe(
+    process.env.NEXT_PUBLIC_STRIPE_PUBLISHABLE_KEY,
```

Comment:
What's the difference between this and `STRIPE_SECRET_KEY`?

Reply:
This [documentation](https://stripe.com/docs/keys) explains it well.